### DISCUSSION TITLE
Need help with creating a scriptlet

### DISCUSSION THREAD

**Discussion by partingscientist** [uBlockOrigin/uBlock-issues#2387] [Q&A] [answered]
I tried writing a scriptlet to redirect Twitter links to a Nitter instance.

[snipped]

I can't seem to make it work. Any possible reason(s) as to why? Thank you in advance.

**Comment by uBlock-user** [comment, marked as answer]
@partingscientist Browser name and version ?

In Firefox, twitter's CSP policy blocks scriptlet execution with the following error as below -- 

![Capture](https://user-images.githubusercontent.com/21290713/204121675-e7f50ece-7611-4fdb-988f-a65733ba71ff.JPG)

**Reply by partingscientist** [reply]
That might be the explanation! I'm using Firefox 107.0. Is there any alternative method to do the redirection?

~~I also forgot to mention something, the filter editor highlighted the nitter word in the filter `twitter.com##+js(nitter)`. I'm pretty sure I've modified the `userResourcesLocation` appropriately and restarted my browser. I'm not sure what's wrong.~~

Addendum: The error part is solved by removing and re-adding the resource location.

**Reply by partingscientist** [reply]
For future reference to self: 
https://bugzilla.mozilla.org/show_bug.cgi?id=1267027

**Comment by mapx-** [comment]
just install scriptlet-doctor:
https://github.com/JustOff/scriptlet-doctor/

then enable it for all sites or only add twitter in his internal list

**Reply by partingscientist** [reply]
Thank you! Will consider this.

**Comment by partingscientist** [comment]
Perhaps I should ask while it's relevant: How do I access arguments in scriptlets?
Say `example.com##+js(foo, arg1, arg2, ...)` as an example.

**Reply by mapx-** [reply]
https://github.com/gorhill/uBlock/blob/master/assets/resources/scriptlets.js

look for {{1}}, {{2}}, .. {{3}} - acs.js for example

parameters acs is waiting:
https://github.com/gorhill/uBlock/wiki/Resources-Library#acsjs-
